src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+        filter: (token) => token.darkValue && token.attributes.category === `color`
+      }]
+    },
+  }
+})
+```
+
+### Graphics
+
+To get the right references in our SVG code we will need to create separate dark SVG files because inside the SVG code we are referencing other design tokens using `.value` but we would need to use `.darkValue` instead. Now our image tokens become:
+
+```json
+{
+  "image": {
+    "logo": {
+      "value": "assets/svg/logo.svg",
+      "darkValue": "assets/svg/logo.dark.svg"
+    },
+    "empty": {
+      "value": "assets/svg/empty.svg",
+      "darkValue": "assets/svg/empty.dark.svg"
+    },
+    "files": {
+      "value": "assets/svg/files.svg",
+      "darkValue": "assets/svg/files.dark.svg"
+    },
+  }
+}
+```
+
+Inside the dark SVG files we reference `.darkValue` instead of `.value`. This is pretty annoying. There is probably a way around this but then we are getting into heavy customizations. 
+
+```xml
+<svg width="100" height="100" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
+  <path d="M5.03278 19C5.54736 11.182 12.0517 5 20 5C27.9482 5 34.4526 11.182 34.9672 19H5.03278Z"
+    fill="<%= color.brand.primary['0'].darkValue %>"/>
+</svg>
+```
+
+In our `generateGraphics` custom action we would see if an token has a `.darkValue` and if it does, generate the dark versions of that graphic for each platform.
+
+```javascript
+// actions/generateGraphics.js
+const { name, value, darkValue } = token;
+//...
+if (darkValue) {
+  const src = template( fs.readFileSync(darkValue) );
+  svgDark = src(dictionary.properties);
+  
+  const outputPath = `${buildPath||''}${name}-dark.svg`;
+  fs.ensureFileSync(outputPath);
+  fs.writeFileSync(outputPath, svgDark);
+  console.log(`✔︎  ${outputPath}`);
+}
+
+androidVector({ androidPath, name, svg, svgDark });
+iosImageset({ iosPath, name, svg, svgDark });
+//...
+```
+
+### Android
+
+We will use roughly the same method as we did for generating CSS files. We will wrap the `android/resources` built-in format with the dark mode format wrapper function and then the `generateResources` action will take care of generating light and dark mode Android vector drawables.
+
+```javascript
+StyleDictionary.extend({
+  ///...
+  platforms: {
+    android: {
+      transformGroup: `android`,
+      buildPath: androidPath,
+      files: [{
+        destination: `values/colors.xml`,
+        format: `android/resources`,
+        filter: (token) => token.attributes.category === `color`,
+        options: {
+          outputReferences: true
+        },
+      },{
+        destination: `values-night/colors.xml`,
+        format: `androidDarkResources`,
+        filter: (token) => token.darkValue && token.attributes.category === `color`
+      }]
+    },
+    //...
+  }
+}).buildAllPlatforms();
+```
+
+### iOS
+
+For iOS we can generate less colorsets and rely on `outputReferences` and custom Swift format to handle referencing the colorsets. 
```

Comment:
For iOS we can generate less colorsets > For iOS, we can generate fewer colorsets